### ISSUE TITLE
Snip not reading Spotify again

### ISSUE THREAD

**Issue by PunZ329** [dlrudie/Snip#315]
I have updated to 6.10.2 back when it was released (6 days ago) and reinstalled it again today (to be on the safe side) but Snip still will not read spotify :/

**Comment by MeikelLP** [comment]
#WorksForMe

Spotify: 1.0.86.337.ga8d5cef9
Snip: v6.10.2
Win10 x64 Home

**Comment by jensz12** [comment]
I have the same issue. Siip will not read whats playing on Spotify

Spotify: 1.0.86.337.ga8d5cef9 (Windows Store version)
Snip: 6.10.2
Windows 10 x64

**Comment by dlrudie** [comment]
Can the windows store version and the desktop version be installed side by side?

**Comment by jensz12** [comment]
No they cant. I found the issue. It works fine, when using the non-windows store version.

**Comment by MeikelLP** [comment]
@dlrudie We need to clarify that only the standalone Spotify works with Snip (in the Readme.md)

**Comment by Jacold** [comment]
Hello. W10 x64, Non-store Spotify version (really needed --enable-audio-graph parameter in target).
Snip says that Spotify isn't running.

**Comment by murko69** [comment]
I'm using the the version (downloaded via [this link](https://download.scdn.co/SpotifySetup.exe)) linked directly from https://www.spotify.com/se/download/windows/ so I'm guessing that's the standalone desktop version?

If so...
Spotify: 1.0.85.259
Snip: 6.10.2
Win 10 Pro x86 (Yes I'm still on 32-bit, but as I understand that should work again?)

Last working Snip for me was 6.9.1 before all the troubles :)
My Snip reads either (Swedish versions of) "changed to Spotify" or plain "spotify isn't running" whenever I use Spotify, and has done for weeks, no matter how many different versions I tried or no matter order I start Spotify/Snip

If this all is due to the store-version, and that's what I'm using after all, then please point me in the direction of correct player and working Snip :)

Thanks!

**Comment by MeikelLP** [comment]
The current implementation is not as stable as we want it to be. I created a [fork](https://github.com/MeikelLP/Snip) (and pull request) not so long ago where Snip uses the official web-based API. This should work without any problems (as far as I tested). If you want you can use this "bloody-edge" version of Snip. But be aware that it's not yet officially accepted by @dlrudie.

**Comment by dlrudie** [comment]
Unfortunately I was away for a week on business. Then I came back for a few days and I’m away again. I’ll be back Thursday and can review all the changes for the implementation then.

**Comment by lownignitus** [comment]
Seems Spotify updated their ToS, and removed support for their Local API. StreamLabs Chatbot had to drop Spotify support because of this it seems.

**Comment by dlrudie** [comment]
Yep. Already known. I bypass any and all apis at the moment but I’ll be switching it up still once I review pull request.

**Comment by lownignitus** [comment]
Awesome, I was here to see if there was an update, saw this, and just wanted to make sure that you had a heads up.

**Comment by PunZ329** [comment]
These are the versions I am on:
Spotify: 1.0.86.337.ga8d5cef9
Snip: 6.10.2
Win10 Pro (64-bit)

As far as I know, I am on the standalone version of Spotify. 

Hopefully this can be resolved soon. I appreciate all you guys do :)

**Comment by murko69** [comment]
Just adding to this issue .
I know I should probably create a new issue, but I just wanted to check if it's a local problem before or if anyone else is actually experiencing this...

Autostarting this app doesn't work for me anymore. I checked, and I still have a shortcut in `{DRIVE}:\Users\{MYUSER}\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup` Last working was the same as with everything else (6.9.1)

**Comment by Berly333** [comment]
I'm not sure if this will help anyone but I found that on my Windows 7 machine that I would get the "Spotify is not currently running" message depending on other windows open at the time I started Snip.  For example, if I have Discord open and start Snip with Spotify already going, it works fine.  However, if I have a Google Chrome browser window open and then try to start Snip with Spotify already running, that's when I get the "Spotify is not currently running" message.  As soon as I close that Chrome window then Snip begins to work correctly.  I didn't try many applications with Snip this way but I did notice it when I had MS Word open.  So this might be a workaround for some.  Start Spotify first, then Snip, then other programs and see if it works?

**Comment by MeikelLP** [comment]
@Berly333 Thanks for this information. I think that's due the insecure implementation as of right now. We have to wait for the Spotify API to be implemented correctly before we can say that it will always work (see #303 for progress).

**Comment by RolllingPanda** [comment]
I have the problem that snip would not let me open the menu to select spotify itunes etc. the right click menu just wouldn´t open has anyone a fix or idea what is wrong?

i have spotify and snip desktop version(not from the windows store)

**Comment by MeikelLP** [comment]
@RolllingPanda Please open a seperate issue for this.

**Comment by AurielTheSun** [comment]
Spotify Version: 1.0.88.353.g15c26ea1
Snip Version: 6.10.2
Windows 7 Home Premium

The Snip.txt document will not write anything in it. Spotify is running and playing music and Snip is running too. Snip has chosen Spotify automatically but the Snip.txt document is jst empty. Pls Help Sincerly Luca

**Comment by lownignitus** [comment]
I am using Snip version 6.10.2 with Spotify Windows Store version 1.0.88.353.g15c26ea1. If I run snip after I have Spotify up and running, then it will not write to the snip.txt, however, if I close Spotify and relaunch it with Snip already running, then I am able to have the currently playing music track in the snip.txt. 

I hope this info is helpful to someone.

**Comment by murko69** [comment]
I'm giving up on this one. Somehow by 'tweaking' betting it to work I mean...
I can't understand why this seems to be working for some ppl and not for the rest (I'm one of the rest).
I have tried the standalone version (that's what I normally use) and also the Windows Store version.
None of them works with Snip.
I have also tried starting one before the other (and vice versa) with no effect what so ever.

How come standalone works for some, Store version for some and starting one before the other for some?
What's the common factor (or uncommon?) here?
Is there any progress made in finding out and fixing this, or is it working for majority of ppl?

**Comment by KritzKast** [comment]
> if I close Spotify and relaunch it with Snip already running, then I am able to have the currently playing music track in the snip.txt.

@lownignitus this is my experience of the issue too. Thanks for finding the work around.

**Comment by tammer123** [comment]
Is there any info when this might be updated?

**Comment by MeikelLP** [comment]
Currently, the creator @dlrudie seems to be inactive. So the chances are pretty low :/

**Comment by dlrudie** [comment]
Sorry, been super tied up with work. I just tested Snip and Spotify here and it's all still working here.

My Spotify version is 1.0.90.268.ga8a0ceb4.

Anyway, I'll try to get on it this weekend and come up with a new solution. I do not like my current solution given the mixed success with using it.

**Comment by DarkTexas** [comment]
Hey there, doesn't work on 1.0.91.183.g259b84fa. 
@dlrudie did you take a look at the changes spotify made? https://developer.spotify.com/community/news/2018/07/19/new-updates-to-spotify-play-button/ 

:(

**Comment by lemba23** [comment]
Same issue with Spotify Version 1.0.92.390.g2ce5ec7d

**Comment by tammer123** [comment]
Does Windows Store Version work at all with Snip right now?

**Comment by GenesisFR** [comment]
Nope, see #263

**Comment by tammer123** [comment]
Yea, I see that, but when I open SNIP and Spotify (Windows Store Version), it does not work. Tried every way, but Snip is not getting any info from Spotify.

**Comment by GenesisFR** [comment]
Snip does NOT work with the Windows Store version.

**Comment by murko69** [comment]
> 
> 
> Sorry, been super tied up with work. I just tested Snip and Spotify here and it's all still working here.
> 
> My Spotify version is 1.0.90.268.ga8a0ceb4.
> 
> Anyway, I'll try to get on it this weekend and come up with a new solution. I do not like my current solution given the mixed success with using it.

Sorry to be a pain in the xxx about this, but is there any hope for Snip and Spotify still? For me it's been ~6 months without a working version (32 or 64 bit) and I miss it dearly!

**Comment by murko69** [comment]
> Snip does NOT work with the Windows Store version.

Could you please offer a direct link to a version that does work? I've tried em all AFAIK but I do still read that Snip+Spotify seems to work for some people. Mine doesn't at all and it's actually been a year since it did for me.

**Comment by GenesisFR** [comment]
I already offered alternatives: https://github.com/dlrudie/Snip/issues/358#issuecomment-495937422

**Comment by murko69** [comment]
Sorry - totally missed that post. Will try them out and get back with feedback
A bit sad to let go of Snip though :/

**Comment by soatok** [comment]
Running both Spotify as administrator, and Snip as administrator, fixed the problem for me. Maybe there's something funky going on with IPC with the latest Windows updates?
